### ISSUE TITLE
[BUG] Incompatible versions of numpy and numba installed by default

### ISSUE THREAD

**Issue by alokito** [chanzuckerberg/cellxgene#2477]
**Describe the bug**
If I pip install Cellxgene, then an error message is printed out
```
ERROR: numba 0.54.1 has requirement numpy<1.21,>=1.17, but you'll have numpy 1.21.5 which is incompatible.
```
Furthermore when I actually run `cellxgene --version` I get an error:
```
cellxgene -version 
Traceback (most recent call last):
  File "/Users/redacted/code/t/testcxg/lib/python3.8/site-packages/pkg_resources/__init__.py", line 583, in _build_master
    ws.require(__requires__)
  File "/Users/redacted/code/t/testcxg/lib/python3.8/site-packages/pkg_resources/__init__.py", line 900, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/Users/redacted/code/t/testcxg/lib/python3.8/site-packages/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (numpy 1.21.5 (/Users/redacted/code/t/testcxg/lib/python3.8/site-packages), Requirement.parse('numpy<1.21,>=1.17'), {'numba'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/redacted/code/t/testcxg/bin/cellxgene", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/Users/redacted/code/t/testcxg/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3251, in <module>
    def _initialize_master_working_set():
  File "/Users/redacted/code/t/testcxg/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3234, in _call_aside
    f(*args, **kwargs)
  File "/Users/redacted/code/t/testcxg/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3263, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/Users/redacted/code/t/testcxg/lib/python3.8/site-packages/pkg_resources/__init__.py", line 585, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/Users/redacted/code/t/testcxg/lib/python3.8/site-packages/pkg_resources/__init__.py", line 598, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/Users/redacted/code/t/testcxg/lib/python3.8/site-packages/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (numpy 1.21.5 (/Users/redacted/code/t/testcxg/lib/python3.8/site-packages), Requirement.parse('numpy<1.21,>=1.17'), {'numba'})

```
**To Reproduce**
1. Using python 3.8, pip install cellxgene

**Expected behavior**
There should be no errors, and cellxgene --version should work

**Version (please complete the following information):**
 - Desktop or hosted?: Desktop
 - Version [e.g. 0.13.0]: 1.0.0

**Additional context**

**Comment by MaximilianLombardo** [comment]
Thanks for your report Alok!

I just tried to reproduce this error in a conda environment with with python 3.8 installed and was unable to. Is there any other context that would be relevant towards reproducing this error?

For reference, I did the following:

```
conda create -n "cellxgene_test" python=3.8.0
conda activate cellxgene_test 
pip install cellxgene
cellxgene --version
```

running `cellxgene --version` gave me the following output in this environment:

`[cellxgene] Version 1.0.0`

Going back to the `pip install` of cellxgene, here is the version of numpy that was installed

```
Collecting numpy>=1.17.5
  Using cached numpy-1.20.3-cp38-cp38-macosx_10_9_x86_64.whl (16.0 MB)
```


Are you able to replicate the original error in a clean conda environment?

**Comment by alokito** [comment]
Thanks for looking into this @MaximilianLombardo ! I went and ran your suggested commands, and it installed cleanly. In my case, the installed versions of numba and numpy were
```
numba-0.54.1 numpy-1.20.3
```

I was able to replicate the original error by using a venv rather than a conda env,
```
python -m venv venv
source venv/bin/activate
pip install cellxgene
cellxgene --version
```
It seems that the issue has to do with how package resolution works within a venv created by miniconda3/bin/python vs  a fresh conda env. For some reason in the venv I am getting numpy 1.22.0 which is incompatible. I noticed that for some reason the venv is using an old version (19.2.3) of pip. Perhaps it could be a pip version issue? I tried the following:

```
python -m venv venv
source venv/bin/activate
pip install --upgrade pip # upgrades pip from 19.2.3 to 21.3.1
pip install cellxgene
cellxgene --version
```
And it worked! So it seems the issue can be solved in venv by upgrading pip. Another mystery solved. Thanks!
